worldengine/draw.py

```diff
@@ -406,6 +407,117 @@ def draw_biome(world, target):
             v = biome[y, x]
             target.set_pixel(x, y, _biome_colors[v])
 
+def draw_scatter_plot(world, size, target):
+    """ This function can be used on a generic canvas (either an image to save
+        on disk or a canvas part of a GUI)
+    """
+
+    #Find min and max values of humidity and temperature on land so we can
+    #normalize temperature and humidity to the chart
+    humid = ma.masked_array(world.humidity['data'], mask=world.ocean)
+    temp = ma.masked_array(world.temperature['data'], mask=world.ocean)
+    min_humidity = numpy.amin(humid)
+    max_humidity = numpy.amax(humid)
+    min_temperature = numpy.amin(temp)
+    max_temperature = numpy.amax(temp)
```

Comment:
Masked arrays themselves have ```min()``` and ```max()``` functions. Might be easier to read, if they do what you need them to do: http://docs.scipy.org/doc/numpy/reference/generated/numpy.ma.MaskedArray.max.html